src/core/instance/events.js

```diff
@@ -66,13 +66,17 @@ export function eventsMixin (Vue: Class<Component>) {
 
   Vue.prototype.$emit = function (event: string): Component {
     const vm: Component = this
-    let cbs = vm._events[event]
+    const camelCaseRE = /([a-z0-9._:])([A-Z])/g
+    let cbs = vm._events[event.toLowerCase()] ||
+              vm._events[event.replace(camelCaseRE, '$1-$2').toLowerCase()]
```

Comment:
there's a utility function for this already, check out: https://github.com/vuejs/vue/blob/v2.0.7/src/shared/util.js#L103